Presentation: Filtering APIs docs (#4959)
iTwin/itwinjs-core: docs/changehistory/NextVersion.md

- [Geometry](#geometry)
  - [Query mesh convexity](#query-mesh-convexity)
- [Write-ahead logging](#write-ahead-logging)
- [Presentation](#presentation)

>   - [Tree hierarchy levels filtering](#tree-hierarchy-levels-filtering)

Reviewer: ```suggestion
  - [Hierarchy levels filtering](#hierarchy-levels-filtering)
```

- Failure to close a writeable briefcase may leave a "-wal" file. Previously, if a program crashed or exited with an open briefcase, it would leave the briefcase file as-of its last call to `IModelDb.saveChanges`. Now, there will be another file with the name of the briefcase with "-wal" appended. This is not a problem and the briefcase is completely intact, except that the briefcase file itself is not sufficient for copying (it will not include recent changes.) The "-wal" file will be used by future connections and will be deleted the next time the briefcase is successfully closed.
- Attempting to copy an open-for-write briefcase file may not include recent changes. This scenario generally only arises for tests. If you wish to copy an open-for-write briefcase file, you must now call [IModelDb.performCheckpoint]($backend) first.

## Presentation

> ### Tree hierarchy levels filtering

Reviewer: ```suggestion
### Hierarchy levels filtering
```

> Ability to filter individual hierarchy level in tree was added for the trees that uses [PresentationTreeDataProvider]($presentation-components). To enable this [PresentationTreeRenderer]($presentation-components) should be passed to [ControlledTree]($components-react) through [ControlledTreeProps.treeRenderer]($components-react):

Reviewer: ```suggestion
Ability to filter individual hierarchy levels was added for tree components that use [PresentationTreeDataProvider]($presentation-components). To enable this, [PresentationTreeRenderer]($presentation-components) should be passed to [ControlledTree]($components-react) through [ControlledTreeProps.treeRenderer]($components-react):
```

```ts
return <ControlledTree
  // other props

>   treeRenderer={(treeProps) => <PresentationTreeRenderer {...treeProps} imodel={props.imodel} modelSource={filteredModelSource} />}

Reviewer: Props aren't very clear (e.g. what is `filteredModelSource`?). Could we just pass `imodel={imodel} modelSoure={modelSource}`?

Author: Copied it straight from test-app did not notices it used those variables. Simplified.

/>;
```

> [PresentationTreeRenderer]($presentation-components) renders nodes with actions buttons for applying and clearing filters. Based on ruleset used in the tree some hierarchy level might not be filterable. In that case actions buttons for those hierarchy levels are not rendered. If applied filter does not produce any nodes `There are no child nodes matching current filter` message is renderer in that hierarchy level.

Reviewer: ```suggestion
[PresentationTreeRenderer]($presentation-components) renders nodes with action buttons for applying and clearing filters. Some hierarchy levels might not be filterable depending on the presentation rules used to build them. In that case, action buttons for those hierarchy levels are not rendered. If applied filter does not produce any nodes, `There are no child nodes matching current filter` message is rendered in that hierarchy level.
```
